From 28d8bdf9434e3553c6fb6eb378f52ebac2413641 Mon Sep 17 00:00:00 2001
From: Brooks Prumo <brooks@solana.com>
Date: Thu, 13 May 2021 17:51:36 -0500
Subject: [PATCH 4/4] Fix clippy (#17214)

Newer clippy from +nightly have new errors about where annotations can and
cannot go.  This commit fixes two of them.
---
 ledger/src/sigverify_shreds.rs | 24 ++++++++++--------------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/ledger/src/sigverify_shreds.rs b/ledger/src/sigverify_shreds.rs
index 6601f07a36d1..34518c6883fd 100644
--- a/ledger/src/sigverify_shreds.rs
+++ b/ledger/src/sigverify_shreds.rs
@@ -244,13 +244,11 @@ pub fn verify_shreds_gpu(
         shred_gpu_offsets(pubkeys_len, batches, recycler_cache);
     let mut out = recycler_cache.buffer().allocate().unwrap();
     out.set_pinnable();
-    elems.push(
-        perf_libs::Elems {
-            #![allow(clippy::cast_ptr_alignment)]
-            elems: pubkeys.as_ptr() as *const solana_sdk::packet::Packet,
-            num: num_packets as u32,
-        },
-    );
+    elems.push(perf_libs::Elems {
+        #[allow(clippy::cast_ptr_alignment)]
+        elems: pubkeys.as_ptr() as *const solana_sdk::packet::Packet,
+        num: num_packets as u32,
+    });
 
     for p in batches {
         elems.push(perf_libs::Elems {
@@ -383,13 +381,11 @@ pub fn sign_shreds_gpu(
     let mut signatures_out = recycler_cache.buffer().allocate().unwrap();
     signatures_out.set_pinnable();
     signatures_out.resize(total_sigs * sig_size, 0);
-    elems.push(
-        perf_libs::Elems {
-            #![allow(clippy::cast_ptr_alignment)]
-            elems: pinned_keypair.as_ptr() as *const solana_sdk::packet::Packet,
-            num: num_keypair_packets as u32,
-        },
-    );
+    elems.push(perf_libs::Elems {
+        #[allow(clippy::cast_ptr_alignment)]
+        elems: pinned_keypair.as_ptr() as *const solana_sdk::packet::Packet,
+        num: num_keypair_packets as u32,
+    });
 
     for p in batches.iter() {
         elems.push(perf_libs::Elems {
-- 
2.31.1

